src/risedevtool/src/config.rs

```diff
@@ -69,62 +73,37 @@ impl ConfigExpander {
         let template_section = global_config
             .get(&Yaml::String("template".to_string()))
             .ok_or_else(|| anyhow!("expect `risedev` section"))?;
-        // selected and expanded profile config.
-        let expanded_config: Vec<(Yaml, Yaml)> = risedev_section
-            .iter()
-            .filter(|(k, _)| k == &&Yaml::String(profile.to_string()))
-            .map(|(profile, v)| {
-                profile
-                    .as_str()
-                    .ok_or_else(|| anyhow!("expect `risedev` section to use string key"))?;
-                let map = v
-                    .as_hash()
-                    .ok_or_else(|| anyhow!("expect `risedev` section to be a hashmap"))?;
 
-                if let Some(config_path) = map.get(&Yaml::String("config-path".to_string())) {
-                    let config_path = config_path
-                        .as_str()
-                        .map(|s| s.to_string())
-                        .ok_or_else(|| anyhow!("expect `config-path` to be a string"))?;
-                    risingwave_config_path = Some(config_path.clone());
-                    update_config(config_path)?;
-                }
+        let profile_section = risedev_section
+            .get(&Yaml::String(profile.to_string()))
+            .ok_or_else(|| anyhow!("profile '{}' not found", profile))?;
+        let profile_map = profile_section
+            .as_hash()
+            .ok_or_else(|| anyhow!("expect `risedev` section to be a hashmap"))?;
 
-                let v = map
-                    .get(&Yaml::String("steps".to_string()))
-                    .ok_or_else(|| anyhow!("expect `steps` section"))?;
-                let mut use_expander = UseExpander::new(template_section)?;
-                let v = use_expander.visit(v.clone())?;
-                let mut dollar_expander = DollarExpander::new(extra_info.clone());
-                let v = dollar_expander.visit(v)?;
-                let mut id_expander = IdExpander::new(&v)?;
-                let v = id_expander.visit(v)?;
-                let mut provide_expander = ProvideExpander::new(&v)?;
-                let v = provide_expander.visit(v)?;
-                Ok::<_, anyhow::Error>((profile.clone(), v))
-            })
-            .try_collect()?;
+        let config_path = profile_map
+            .get(&Yaml::String("config-path".to_string()))
+            .and_then(|s| s.as_str())
+            .unwrap_or("src/config/risingwave.toml")
```

Comment:
Well if the values in `config/risingwave.toml` is the same as defaults in code, and we want to use it as a sample, we can simply avoid using it completely? 🤔

Reply:
Look good to me. So `risingwave.toml` will be useless at all.

Reply:
Oops, that would require me to revert `String` back to `Option<String>` 🤣

Reply:
Updated. Please take a look @xxchan